source/Parser_expressions.cpp

```diff
@@ -33,7 +33,7 @@ ExpressionSyntax* Parser::parseSubExpression(ExpressionOptions::Enum options, in
     int newPrecedence = 0;
 
     auto current = peek();
-    if (current.kind == TokenKind::NewKeyword)
+    if (current.kind == TokenKind::NewKeyword || isClassScope(TokenKind::NewKeyword))
```

Comment:
Rather than having to scan here for class scope, I think we should also check for new expressions in the postfix phase, at which point the class scope will already have been parsed into a NameSyntax.